### PR TITLE
templates: Enable android builds

### PR DESCRIPTION
**Pull request by fetzerch**
Related to: https://github.com/kodi-game/game.libretro.2048/pull/1

Jenkins build: http://jenkins.kodi.tv/view/Helpers/job/BuildMulti-All/1818/

### CODE REVIEW

File: templates/addon/depends/common/{{ game.name }}/CMakeLists.txt.j2 [hunk 1]
```diff
@@ -22,13 +22,29 @@ elseif(CORE_SYSTEM_NAME STREQUAL linux)
...
+  elseif(CPU STREQUAL android_x86)
+    set(PLATFORM android-x86)
+  else()
+    message(FATAL_ERROR "${PROJECT_NAME} needs Android ${CPU} build command in CMakeLists.txt!")
+  endif()
+  set(BUILD_COMMAND $(MAKE) -C {{ makefile.dir }} -f {{ makefile.file }} {{ config.cmake_options | default('') }} platform=${PLATFORM} ${LIBRETRO_DEBUG})
 elseif(CORE_SYSTEM_NAME STREQUAL rbpi)
-  message(FATAL_ERROR "${PROJECT_NAME} needs RPi build command in CMakeLists.txt!")
+  if(CPU STREQUAL arm1176jzf-s)
+    set(PLATFORM rpi2)
```
Comment (garbear) [inline templates/addon/depends/common/{{ game.name }}/CMakeLists.txt.j2]: I thought the rpi1 was armv6?

File: templates/addon/depends/common/{{ game.name }}/CMakeLists.txt.j2 [hunk 2]
```diff
@@ -22,13 +22,29 @@ elseif(CORE_SYSTEM_NAME STREQUAL linux)
...
+  elseif(CPU MATCHES "cortex-a7" OR CPU MATCHES "cortex-a53")
+    set(PLATFORM rpi3)
+  else()
+    message(FATAL_ERROR "${PROJECT_NAME} needs Rbpi ${CPU} build command in CMakeLists.txt!")
+  endif()
+  set(BUILD_COMMAND $(MAKE) -C {{ makefile.dir }} -f {{ makefile.file }} {{ config.cmake_options | default('') }} platform=${PLATFORM} ${LIBRETRO_DEBUG})
 elseif(CORE_SYSTEM_NAME STREQUAL freebsd)
-  message(FATAL_ERROR "${PROJECT_NAME} needs FreeBSD build command in CMakeLists.txt!")
+  set(BUILD_COMMAND make -f Makefile.libretro platform=bsd ${LIBRETRO_DEBUG})
```
Comment (garbear) [inline templates/addon/depends/common/{{ game.name }}/CMakeLists.txt.j2]: missing cmake parameters?

**Comment by garbear** [comment]
iOS might have failed because the compiled libraries end in .so instead of .dylib. I think this was an issue for OSX once

**Comment by fetzerch** [comment]
I think it needs testing locally. Will try to do that on the weekend.

**Comment by fetzerch** [comment]
Next attempt: http://jenkins.kodi.tv/view/Helpers/job/BuildMulti-All/1821/

**Comment by fetzerch** [comment]
I'll focus on android for the moment, unfortunately it seems that there are multiple ways to build the cores.
Most cores use ndk-build. 

Jenkins: http://jenkins.kodi.tv/job/Android-ARM/10437/

RBPi is difficult. Some cores use platform=rpi,rpi2,rpi3 others use platform=*arm*.

**Comment by fetzerch** [comment]
It's alive! First droid build:
- http://jenkins.kodi.tv/job/Android-ARM/10438/
- http://mirrors.kodi.tv/test-builds/android/arm/kodi-20161223-ad1b13a-script-test-armeabi-v7a.apk

still open:
- x86, aarch64 builds (afaik the libretro project doesn't yet do aarch64 builds. curious if at least some cores will just build).
- few missing cores that use a slightly other way of compiling

**Comment by fetzerch** [comment]
First working(!) droid builds:
- http://jenkins.kodi.tv/job/Android-ARM/10442/
- http://jenkins.kodi.tv/job/Android-ARM64-v8a/934
- http://jenkins.kodi.tv/job/Android-X86/9322/

I'll merge this now and push changes to the game addons.

**Pull request merged**